duckduckgo/iOS
# ipad improvements

Task/Issue URL: 
* Top level task & Figma link https://app.asana.com/0/392891325557410/874480958064197 
* Implementation detail https://app.asana.com/0/392891325557410/1184791777204513

**Description**:

Improves iPad experience by adding:
* tabs
* stylised 'suggestions' tray 
* stylised favicons (grey box with white text letter)
* tweaked keyboard commands (hold Cmd to see the options, except for the arrow keys which aren't shown there but should still behave in a non-surprising way)
* better pointer support for keyboard users - this may be difficult to test without the physical keyboard for iPad peripheral 

Key concepts:

* Only users in the experiment should see any noticeable change to the UI 
* Only users on an iPad can be in the experiment
* Everyone else should see the iPhone experience on both the iPhone and the iPad
* If in doubt - see what Safari does in the same situation we should be fairly close to Safari where it makes sense

Also adds the following changes:

* Improved variant filtering
* Updated dax dialog copy
* Improved accessibility options for some buttons and rationalisation of some copy across the fire buttons
* Removes some unused code (e.g. App Review prompt in feedback)

**Steps to test this PR**:
1. Migrate from existing build - should not be added to the experiment
1. Fresh install until you get the experiment variant (or hack the code)
1. Search, browse, add favorites, add bookmarks, etc
1. Play with tabs; adding, reordering and making sure they reflect changes made elsewhere (tab switcher)
1. Use the app in "slide over" mode - it should have the "small" interface (no tabs, bottom button bar)
1. Use the app in "thin" side by side mode - it should have the iPhone interface (no tabs, bottom button bar)
1. Use the app in any of the "larger" side by side modes - it should have the larger interface (tabs, no bottom button bar)
 * note: portrait orientation on the iPad maybe surprising in that more of the screen is considered 'small' - again, check vs Safari



**Orientation Testing**:


**Device Testing**:

* [x] iPad 12.9"
* [ ] iPad mini 

**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
